tests/nnstreamer_query/unittest_query.cc

```diff
@@ -0,0 +1,254 @@
+  g_object_get (srv_handle, "port", &uint_val, NULL);
+  EXPECT_EQ (5000U, uint_val);
+
+  g_object_set (srv_handle, "protocol", 1, NULL);
+  g_object_get (srv_handle, "protocol", &int_val, NULL);
+  EXPECT_EQ (1, int_val);
+
+
+  g_object_set (srv_handle, "timeout", 20U, NULL);
+  g_object_get (srv_handle, "timeout", &uint_val, NULL);
+  EXPECT_EQ (20U, uint_val);
+
+  gst_object_unref (srv_handle);
+  gst_object_unref (gstpipe);
+  g_free (pipeline);
+}
+
+/**
+ * @brief Test for tensor_query_server with same port.
+ */
+TEST (tensorQuery, serverProperties1_n)
+{
+  gchar *pipeline;
+  GstElement *gstpipe;
+
+  /* Create a nnstreamer pipeline */
+  pipeline = g_strdup_printf (
+      "tensor_query_serversrc name=serversrc port=3000 ! "
+      "other/tensors,num_tensors=1,dimensions=3:300:300:1,types=uint8 ! "
+      "tensor_query_serversink port=3000 sync=false async=false");
+  gstpipe = gst_parse_launch (pipeline, NULL);
+  EXPECT_NE (pipeline, nullptr);
+
+  EXPECT_NE (setPipelineStateSync (gstpipe, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+
+  gst_object_unref (gstpipe);
+  g_free (pipeline);
+}
+
+/**
+ * @brief Test for tensor_query_client get and set properties
+ */
+TEST (tensorQuery, clientProperties0)
+{
+  gchar *pipeline;
+  GstElement *gstpipe;
+  GstElement *client_handle;
+  gint int_val;
+  guint uint_val;
+  gchar *str_val;
+  gboolean bool_val;
+
+  /* Create a query client pipeline */
+  pipeline = g_strdup_printf (
+      "videotestsrc ! videoconvert ! videoscale ! video/x-raw,width=300,height=300,format=RGB !"
+      "tensor_converter ! tensor_query_client name=client protocol=tcp src-host=127.0.0.1 sink-host=127.0.0.1 ! tensor_sink");
+  gstpipe = gst_parse_launch (pipeline, NULL);
+  EXPECT_NE (gstpipe, nullptr);
+
+  /* Get properties of query client */
+  client_handle = gst_bin_get_by_name (GST_BIN (gstpipe), "client");
+  EXPECT_NE (client_handle, nullptr);
+
+  g_object_get (client_handle, "src-host", &str_val, NULL);
+  EXPECT_STREQ ("127.0.0.1", str_val);
+  g_free (str_val);
+
+  g_object_get (client_handle, "src-port", &uint_val, NULL);
+  EXPECT_EQ (3001U, uint_val);
+
+  g_object_get (client_handle, "sink-host", &str_val, NULL);
+  EXPECT_STREQ ("127.0.0.1", str_val);
+  g_free (str_val);
+
+  g_object_get (client_handle, "sink-port", &uint_val, NULL);
+  EXPECT_EQ (3000U, uint_val);
+
+  g_object_get (client_handle, "protocol", &str_val, NULL);
+  EXPECT_STREQ ("tcp", str_val);
+  g_free (str_val);
+
+  g_object_get (client_handle, "silent", &bool_val, NULL);
+  EXPECT_EQ (TRUE, bool_val);
+
+  /* Set properties of query client */
+  g_object_set (client_handle, "src-host", "127.0.0.2", NULL);
+  g_object_get (client_handle, "src-host", &str_val, NULL);
+  EXPECT_STREQ ("127.0.0.2", str_val);
+  g_free (str_val);
+
+  g_object_set (client_handle, "src-port", 5001U, NULL);
+  g_object_get (client_handle, "src-port", &int_val, NULL);
+  EXPECT_EQ (5001U, int_val);
```

Comment:
Is it necessary? If so, you should fix build error.
Comparison signed and unsigned value makes build error on app-test (e.g., line 190, 199)

Reply:
No. My bad. Updated.